src/smem.c

```diff
@@ -37,23 +38,21 @@ bool ice_smem_is_async_complete(void) {
     return g_cs_pin < 0;
 }
 
-void ice_smem_set_async_callback(ice_smem_async_callback_t callback) {
-    g_async_callback = callback;
-}
+/// Select the serial memory for SPI transaction.
+static void smem_select(int cs_pin) {
+    assert(g_cs_pin < 0);
 
-/// Select the SRAM for SPI transaction.
-static void ice_smem_select(int cs_pin) {
     g_cs_pin = cs_pin;
 
     // Short delay for two reasons: 1) to pull-up SSRAM CS via pull-up resistor and
     // 2) to ensure memory is deselected long enough to refresh DRAM.
-    sleep_us(1);
+    busy_wait_us(1);
```

Comment:
sleep_us() -> busy_wait_us() because former may not be called from an interrupt handler